Makefile

```diff
@@ -101,9 +101,11 @@ manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and Cust
 
 deps: controller-gen ## Download all dependencies if necessary.
 
-CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
-controller-gen: ## Download controller-gen locally if necessary.
-	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.9.0)
+CONTROLLER_GEN = $(GOBIN)/controller-gen
+controller-gen: $(CONTROLLER_GEN) ## Download controller-gen locally if necessary.
+
+$(CONTROLLER_GEN):
+	go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.8.0
```

Comment:
Why not
```
controller-gen:
    go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.8.0
```

Reply:
In this case, `go install` will run regardless of whether the binary exists. Do we want to run the command it on all invocations?

Reply:
I am running `make controller-gen` as an example here and I don't see it getting installed every time and that's the behaviour I would expect

Reply:
With GNU make and the following rule, the `go install` is executed for every incovation of `make controller-gen` or any other rule that has controller-gen as a dependency. 

```
CONTROLLER_GEN = $(shell pwd)/bin/controller-gen                                                                                                                                                                              
controller-gen:                                                                                                                                                                                                                       
         go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.8.0
```

And it is the expected behavior according to the documentation:
- https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html

I also see it consistently happening on my system.

The command will not run if there is the `controller-gen` file present the current directory. Or maybe with a different sort of `make`. Could it be your case?

However, after experimenting with the rule and `go install`, your suggestion might work the best (with a small optimization). It is better to run `go install` always to ensure the right version. Otherwise, if it exists on a local repo checkout and is of different version, it will not be updated and may cause discrepancies. Moreover, `go install` is cheap and fast for the subsequent executions thanks to deps being downloaded.